packages/synapse-react-client/src/components/DataGrid/useDataGridWebsocket.ts

```diff
@@ -280,15 +286,30 @@ export function useDataGridWebSocket() {
     }
   }, [state.websocketInstance])
 
+  /**
+   * Checks if the model snapshot contains the minimum data required for rendering (columns and rows).
+   */
+  function isModelRenderable(model: GridModel | null) {
+    if (!model || !model.api.getSnapshot() || !modelSnapshot) {
+      return false
+    }
+    const { columnNames, columnOrder, rows } = modelSnapshot
+    const columnsReady = columnNames.length >= 1
+    const orderReady = columnOrder.length >= 1
+    const rowsReady = rows.length >= 1
```

Comment:
I think it might be possible to have a RecordSet with 0 rows, so let's allow it for now.
```suggestion
    const rowsReady = rows.length >= 0
```